pkg/kv/kvserver/replica_raft.go

```diff
@@ -3091,9 +3096,21 @@ func (r *Replica) updateLastUpdateTimesUsingStoreLivenessRLocked(
 	for _, desc := range r.descRLocked().Replicas().Descriptors() {
 		// If the replica's store if providing store liveness support, update
 		// lastUpdateTimes to indicate that it is alive.
+		//
+		// Otherwise, mark the replica as unreachable, so that eventually raft is
+		// notified by ReportUnreachable, and the flow transfers to StateProbe. Do
+		// this only the first time, so that addUnreachableRemoteReplica is called
+		// only once per disconnect.
+		//
+		// TODO(pav-kv): there is an overlap between r.mu.lastUpdateTimes and
+		// r.unreachablesMu. Unify them into one mechanism.
 		_, curExp := (*replicaRLockedStoreLiveness)(r).SupportFrom(raftpb.PeerID(desc.ReplicaID))
 		if storeClockTimestamp.ToTimestamp().LessEq(curExp) {
 			r.mu.lastUpdateTimes.update(desc.ReplicaID, r.Clock().PhysicalTime())
+		} else if upd := r.mu.lastUpdateTimes[desc.ReplicaID]; !upd.unreachable {
+			upd.unreachable = true
+			r.mu.lastUpdateTimes[desc.ReplicaID] = upd
```

Comment:
This update might be problematic because `Replica.mu` is only locked for reads.

Reply:
Btw, the above `r.mu.lastUpdateTimes.update` is incorrect for the same reason.

Reply:
Luckily, `mu` is already locked for writes. So I just "fixed" this by renaming the method in another commit.